### PR TITLE
[MLDB-1266] Added import json procedure

### PR DESCRIPTION
**Pull request by mailletf**

### CODE REVIEW

File: container_files/assets/www/doc/builtin/procedures/JSONImporter.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    {"a": 5, "b": true}
+
+is recorded as:
+
+| *rowName* | *a* | *b* |
+|-----------|-----|-----|
+| row1 | 5 | true |
+
+If the value is an object, we apply the same logic recursively, adding an underscore
```
Comment (guyd) [inline container_files/assets/www/doc/builtin/procedures/JSONImporter.md]: Are we using period '.' in some other cases to flatten a nested object?  If so, we may want to use the same separator.

File: container_files/assets/www/doc/builtin/procedures/JSONImporter.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+is recorded as:
+
+| *rowName* | *a* | *b* | *c_x* |
+|-----------|-----|-----|-------|
+| row1 | 5 | true | hola |
+
+
+If the value is an array that contains only atomic types (strings, bool or numeric), we
+encode them as one hot vectors.
```
Comment (guyd) [inline container_files/assets/www/doc/builtin/procedures/JSONImporter.md]: What is a _hot vector_?

File: plugins/json_importer.cc [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+DECLARE_STRUCTURE_DESCRIPTION(JSONImporterConfig);
+
+DEFINE_STRUCTURE_DESCRIPTION(JSONImporterConfig);
+
+JSONImporterConfigDescription::
+JSONImporterConfigDescription()
+{
+    addField("dataFileUrl", &JSONImporterConfig::dataFileUrl,
+             "URL to load text file from");
+    addField("output", &JSONImporterConfig::output,
```
Comment (guyd) [inline plugins/json_importer.cc]: output -> outputDataset

File: plugins/json_importer.cc [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+             "Configuration for output dataset",
+             PolyConfigT<Dataset>().withType("sparse.mutable"));
+    addField("limit", &JSONImporterConfig::limit,
+             "Maximum number of lines to process");
+    addField("offset", &JSONImporterConfig::offset,
+            "Skip the first n lines (excluding the header if present).", int64_t(0));
```
Comment (guyd) [inline plugins/json_importer.cc]: What would be a header here?

File: plugins/json_importer.cc [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+    addField("ignoreBadLines", &JSONImporterConfig::ignoreBadLines,
+             "If true, any line causing an error will be skipped.", false);
```
Comment (guyd) [inline plugins/json_importer.cc]: Are json objects with missing values considered an error?  Not sure if this option is well-defined for json.

File: plugins/json_importer.cc [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+            
+            recordedLines++;
+
+            std::unique_lock<std::mutex> guard(recordMutex);
+            output->recordRow(outputRow.rowName, outputRow.columns);
+            return true;
+        };
+
+
+        forEachLineStr(runProcConf.dataFileUrl.toString(),
```
Comment (guyd) [inline plugins/json_importer.cc]: The csv importer was changed to use forEachLineBlock for better performance.  Consider using the same here.

**Comment by guyd** [comment]
Added minor comments.  Consider also adding one test case with an invalid json.  +1

File: container_files/assets/www/doc/builtin/procedures/JSONImporter.md (hunk 1 above)
Comment (jeremybarnes) [inline container_files/assets/www/doc/builtin/procedures/JSONImporter.md]: We should use periods.

**Pull request merged**